integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -451,11 +462,17 @@ async def get_group(self, group_id: int) -> Group | None:
         else:
             return None
 
-    async def get_all_groups(self) -> typing.AsyncIterator[List[Group]]:
+    @cache_iterator_result()
```

Comment:
this isn't consistent, on one hand you are using `@cache_iterator_result()` and for `users` you are specifically creating cache. Ideally lets use `cache_iterator_result` if possible

Reply:
`cache_iterator_results` works with generators, not coroutines...
I wrote one for coroutines [here](https://github.com/port-labs/ocean/blob/a7e4aed2b9a312b145121e6bd4f29783f37e749b/integrations/gcp/gcp_core/cache.py#L9), I'll push to ocean core utils so we use it from there

Reply:
[Here](https://github.com/port-labs/ocean/pull/977)